src/pw/pw_methods.F

```diff
@@ -2186,7 +2259,11 @@ SUBROUTINE pw_smoothing(pw, ecut, sigma)
                      DO ig = 1, cnt
                         arg = (ecut - pw%pw_grid%gsq(ig))/sigma
                         f = EXP(arg)/(1 + EXP(arg))
+#if defined(__GNUC__) && defined(__GNUC__MINOR__) && CPVERSION_CHECK(9, 5, >, __GNUC__, __GNUC__MINOR__)
+                        pw%array(ig) = CMPLX(f*REAL(pw%array(ig), KIND=dp), f*AIMAG(pw%array(ig)), KIND=dp)
```

Comment:
I am not sure. There was a recent report in the Google group with a similar issue. This is why I have decided to rewrite similar parts of the pw_methods module in a similar more explicitly manner. It may work standard-wise but the Intel compiler complains in certain circumstances. I will double-check the code as I can reproduce the original issue locally.

Reply:
Is it worth using conditional compilation just to use `%re` and `%im` as opposed to `REAL` and `AIMAG`? The test `defined(__GNUC__) && defined(__GNUC__MINOR__)` is actually part of  `CPVERSION_CHECK` and can be dropped - so, things get surprisingly short ;-).

Reply:
After double-checking, we have probably hit some bug in the compiler. I am not sure what exactly caused it but replacing the `WORKSHARE`-construct with a `DO`-loop and rewriting the three steps as one, should have done the job, too.